src/os/interaction/modifyinteraction.js

```diff
@@ -0,0 +1,258 @@
+goog.module('os.interaction.Modify');
+goog.module.declareLegacyNamespace();
+
+const Circle = goog.require('ol.style.Circle');
+const Feature = goog.require('ol.Feature');
+const Fill = goog.require('ol.style.Fill');
+const I3DSupport = goog.require('os.I3DSupport');
+const KeyCodes = goog.require('goog.events.KeyCodes');
+const KeyHandler = goog.require('goog.events.KeyHandler');
+const OLModify = goog.require('ol.interaction.Modify');
+const PayloadEvent = goog.require('os.events.PayloadEvent');
+const Point = goog.require('ol.geom.Point');
+const RecordField = goog.require('os.data.RecordField');
+const Stroke = goog.require('ol.style.Stroke');
+const Style = goog.require('ol.style.Style');
+const osImplements = goog.require('os.implements');
+const {MODAL_SELECTOR} = goog.require('os.ui');
+const {ModifyEventType} = goog.require('os.interaction');
+const {notifyStyleChange} = goog.require('os.style');
+const osWindow = goog.require('os.ui.window');
+const windowSelector = goog.require('os.ui.windowSelector');
+
+const KeyEvent = goog.requireType('goog.events.KeyEvent');
+const OSMap = goog.requireType('os.Map');
+
+
+/**
+ * Allows the user to modify geometries on the map directly.
+ *
+ * @implements {I3DSupport}
+ */
+class Modify extends OLModify {
+  /**
+   * Constructor.
+   * @param {olx.interaction.ModifyOptions=} opt_options Options.
+   *
+   * @suppress {accessControls}
+   */
+  constructor(opt_options) {
+    const options = opt_options || {};
+
+    super(options);
+
+    /**
+     * @type {KeyHandler}
+     * @protected
+     */
+    this.keyHandler = new KeyHandler(document, true);
+
+    this.keyHandler.listen(KeyHandler.EventType.KEY, this.handleKeyEvent, true, this);
+
+    // jank alert: the functions that are called when the interaction starts and ends are hard to override, so instead
+    // listen to our own events and toggle the map movement on and off
+    ol.events.listen(this, ol.interaction.ModifyEventType.MODIFYSTART, this.handleStart, this);
+    ol.events.listen(this, ol.interaction.ModifyEventType.MODIFYEND, this.handleEnd, this);
+
+    // use the drawing layer as the overlay layer so that geometries are synced to WebGL automatically
+    this.overlay_ = os.map.mapContainer.getDrawingLayer();
```

Comment:
Nit: missing require() and .getInstance()

Reply:
`ol.events` and `ol.interaction.ModifyEventType` should also be referenced by require.

Reply:
I'll add a comment here. Requiring the `MapContainer` was creating a circular dependency somewhere, which was why I went to the sucky reference. I take a quick look to see if I can untangle it, but that's a big mess for later.